### PR TITLE
planner: make (*AccessPath).OnlyPointRange more succinct

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

make `(*AccessPath).OnlyPointRange` more succinct

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/util/path.go [hunk 1]
```diff
@@ -151,35 +151,21 @@ func isColEqCorColOrConstant(ctx sessionctx.Context, filter expression.Expressio
 
 // OnlyPointRange checks whether each range is a point(no interval range exists).
 func (path *AccessPath) OnlyPointRange(sctx sessionctx.Context) bool {
```
Comment (qw4990) [inline planner/util/path.go:153]: Please add some UT for this function.

File: planner/util/path.go [hunk 2]
```diff
@@ -151,35 +151,21 @@ func isColEqCorColOrConstant(ctx sessionctx.Context, filter expression.Expressio
...
-	noIntervalRange := true
 	if path.IsIntHandlePath {
 		for _, ran := range path.Ranges {
-			if !ran.IsPoint(sctx) {
-				noIntervalRange = false
-				break
+			if !ran.IsPointNonNullable(sctx) {
```
Comment (qw4990) [inline planner/util/path.go:156]: Can we use IsPointNullable here?

File: planner/util/path.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply planner/util/path.go:153]: Added.

File: planner/util/path.go (hunk 2 above)
Comment (xuyifangreeneyes) [reply planner/util/path.go]: Done.

**Review by qw4990 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**